sqlglot/transforms.py

```diff
@@ -105,7 +105,16 @@ def eliminate_qualify(expression: exp.Expression) -> exp.Expression:
                 select.replace(exp.alias_(select, alias))
                 taken.add(alias)
 
-        outer_selects = exp.select(*[select.alias_or_name for select in expression.selects])
+        def _is_quoted(select: exp.Expression) -> bool:
+            _ident = select.args.get("alias") or select.this
+            return _ident.args.get("quoted") if _ident else False
```

Comment:
This was intentionally returning an optional bool, the `quoted` arg differentiates between the `None` and `False` cases:
```suggestion
        def _is_quoted(select: exp.Expression) -> t.Optional[bool]:
            return (select.args.get("alias") or select.this).args.get("quoted")
```